### DISCUSSION TITLE
Limit on used cores

### DISCUSSION THREAD

**Discussion by KavehNNejad** [NREL/REopt-API-Analysis#49] [8 - API: Application Programming Interface] [answered]
While running some scenarios, I noticed that the docker container would not exceed 1.4 cores used, and rarely went above 1, even when assigned 12 cores and multiple API calls were made simultaneously.

I was curious if this is expected, or if I should investigate more? as that would effect the number of cores needed on the solver license.

I am unsure if this would affect it, but just in case, we are using a test licence provided by FICO, although I cannot see any limit on cores in the licence.

**Comment by NLaws** [comment, marked as answer]
The most cores used by any solver is typically during the presolve phase (if it is allowed to parallelize). However, if you are submitting multiple jobs at once and you have multiple celery and julia containers running then you can expect to see more cores being used, commensurate with your number of container (or pods in kubernetes). If you only have one julia and one celery container then I would not expect to see more than two cores being utilized at any given time. Note that you can also adjust the number of threads that any julia pod can use when you build the julia container (https://github.com/NREL/REopt_Lite_API/blob/f48d93777eaaad2f62587cc70a6241314d215750/julia_src/Dockerfile#L28). So, I'm guessing that you only have one julia container running, as well as one celery container. The multiple jobs are sitting in the redis queue waiting for the celery and julia containers to be available.
